lib/protobuf/nats/client.rb

```diff
@@ -13,8 +14,85 @@ def close_connection
         # no-op (I think for now), the connection to server is persistent
       end
 
+      def self.subscription_key_cache
+        @subscription_key_cache ||= {}
+      end
+
       def send_request
-        # after method execution the "response" should be in @response_data and should be a string of bytes
+        retries ||= 3
```

Comment:
👍 I agree. I put the retry in place to match zmq as closely as possible. I'm happy to retry only when we don't get an ACK from a server. Thoughts?